storage_controller/src/persistence.rs

```diff
@@ -572,6 +572,30 @@ impl Persistence {
             .collect())
     }
 
+    /// Read the generation number of a specific tenant shard
+    pub(crate) async fn get_generation(
```

Comment:
nit: Could avoid the `get_` here as general naming issue.

Reply:
It could be find_generations

Reply:
name changed in 869dfde8c82f6569415478d81c2ad7e9a7889b41.

I'm not sure if we need to apply the "omit the get_" style point to functions which aren't simple getters, but in this case it made sense to go with shard_generations to disambiguate it from the method that gets all the generations for a tenant's shards.